test/shell.mjs

```diff
@@ -89,6 +113,117 @@ describe("shell", function () {
     it.skip("can read input from a file", async function () {});
 
     it.skip("can set a connection timeout", async function () {});
+
+    const upArrow = "\x1b[A";
+    const downArrow = "\x1b[B";
+
+    it("can keep track of history", async function () {
+      // start the shell
+      const runPromise = run(`shell --secret "secret" --typecheck`, container);
```

Comment:
i think we should wrap these in a describe block - this lets you group them, and lets you register a beforeEach hook that can unstub `fs`.

if you look at a test run report, you can see how the spec names are built from describe / it block traversals:
![image](https://github.com/user-attachments/assets/0861a047-2b75-4d01-84d7-c922b88ec538)

given that, i think these tests should live somewhere like:
```
shell > common > history > works in one session
... > can be cleared
... > is saved between sessions
```

ideally, the test path builds a readable sentence.

Reply:
😁